test/e2e/apimachinery/webhook.go

```diff
@@ -703,8 +704,381 @@ var _ = SIGDescribe("AdmissionWebhook [Privileged:ClusterAdmin]", func() {
+) *admissionregistrationv1.ValidatingWebhookConfiguration {
+	sideEffects := admissionregistrationv1.SideEffectClassNone
+	equivalent := admissionregistrationv1.Equivalent
+	return &admissionregistrationv1.ValidatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: f.UniqueName,
+		},
+		Webhooks: []admissionregistrationv1.ValidatingWebhook{
+			{
+				Name: "validation-webhook-with-match-conditions.k8s.io",
+				Rules: []admissionregistrationv1.RuleWithOperations{{
+					Operations: []admissionregistrationv1.OperationType{admissionregistrationv1.Create},
+					Rule: admissionregistrationv1.Rule{
+						APIGroups:   []string{""},
+						APIVersions: []string{"v1"},
+						Resources:   []string{"*"},
+					},
+				}},
+				ClientConfig: admissionregistrationv1.WebhookClientConfig{
+					Service: &admissionregistrationv1.ServiceReference{
+						Namespace: f.Namespace.Name,
+						Name:      serviceName,
+						Path:      strPtr("/always-deny"),
+						Port:      pointer.Int32(servicePort),
+					},
+					CABundle: certCtx.signingCert,
+				},
+				SideEffects:             &sideEffects,
+				MatchPolicy:             &equivalent,
+				AdmissionReviewVersions: []string{"v1"},
+				// Scope the webhook to just the markers namespace
+				NamespaceSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{f.UniqueName: "true"},
+				},
+				MatchConditions: matchConditions,
+			},
+			newValidatingIsReadyWebhookFixture(f, certCtx, servicePort),
+		},
+	}
+}
+
+func newMutatingWebhookWithMatchConditions(
+	f *framework.Framework,
+	servicePort int32,
+	certCtx *certContext,
+	matchConditions []admissionregistrationv1.MatchCondition,
+) *admissionregistrationv1.MutatingWebhookConfiguration {
+	sideEffects := admissionregistrationv1.SideEffectClassNone
+	return &admissionregistrationv1.MutatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: f.UniqueName,
+		},
+		Webhooks: []admissionregistrationv1.MutatingWebhook{
+			{
+				Name: "adding-configmap-data.k8s.io",
+				Rules: []admissionregistrationv1.RuleWithOperations{{
+					Operations: []admissionregistrationv1.OperationType{admissionregistrationv1.Create},
+					Rule: admissionregistrationv1.Rule{
+						APIGroups:   []string{""},
+						APIVersions: []string{"v1"},
+						Resources:   []string{"configmaps"},
+					},
+				}},
+				ClientConfig: admissionregistrationv1.WebhookClientConfig{
+					Service: &admissionregistrationv1.ServiceReference{
+						Namespace: f.Namespace.Name,
+						Name:      serviceName,
+						Path:      strPtr("/mutating-configmaps"),
+						Port:      pointer.Int32(servicePort),
+					},
+					CABundle: certCtx.signingCert,
+				},
+				SideEffects:             &sideEffects,
+				AdmissionReviewVersions: []string{"v1", "v1beta1"},
+				// Scope the webhook to just this namespace
+				NamespaceSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{f.UniqueName: "true"},
```

Comment:
I don't think the namespace actually has this unique label, at least based on https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/framework.go#L261.

What you want here instead is:
```
MatchLabels: map[string]string{v1.LabelMetadataName: f.UniqueName},
```

Reply:
It's these functions that require those specific matchLabels https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apimachinery/webhook.go#L2201-L2226 
From my understanding, it's used to avoid collision with the other e2e tests.

Reply:
Right, but the namespaces themselves have to have this label right?

Reply:
Call is here https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apimachinery/webhook.go#L95-L96
Labeling function is here https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apimachinery/webhook.go#L2056-L2068

Reply:
Ah I see, thanks for the reference